core/node/node_framework/src/implementations/layers/eth_watch.rs

```diff
@@ -66,21 +74,54 @@ impl WiringLayer for EthWatchLayer {
         let main_pool = input.master_pool.get().await?;
         let client = input.eth_client.0;
 
-        let eth_client = EthHttpQueryClient::new(
+        let sl_diamond_proxy_addr = if self.settlement_mode.is_gateway() {
+            self.gateway_chain_config
+                .clone()
+                .context("Lacking `gateway_contracts_config`")?
+                .diamond_proxy_addr
+        } else {
+            self.contracts_config.diamond_proxy_addr
+        };
+        tracing::info!(
+            "Diamond proxy address ethereum: {}",
```

Comment:
Nit: IIUC, `Display` representation for `Address` and other fixed hashes like `H256` doesn't output all hex digits, but rather only starting 2 and ending 2 digits. Use the `Debug` repr (`{:?}`) to output the full address.